### PR TITLE
Add SonarLint exclusion/inclusion IT

### LINKED ISSUE #6811: Add IT to verify the exclusion-inclusion logic works properly
It should be possible to cover this functionality in the .NET analyzer ITs. Add a new project with multiple files, and add inclusion/exclusion patterns to include or exclude certain files, and make sure that issue are reported only for files that are included.

### PR DESCRIPTION
**Pull request by mary-georgiou-sonarsource**
Fixes #6811

### CODE REVIEW

File: analyzers/its/sources/SonarLintExclusions/SonarLintExclusions/NotIncluded.cs [hunk 1]
```diff
@@ -0,0 +1,12 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace SonarLintExclusions
+{
+    internal class NotIncluded
```
Comment (csaba-sagi-sonarsource) [inline analyzers/its/sources/SonarLintExclusions/SonarLintExclusions/NotIncluded.cs:9]: Could you please align it with the other files? Remove the usings and add the  // S2094 comment

File: analyzers/its/sources/SonarLintExclusions/SonarLintExclusionsTest/SonarLintExclusionsTest.csproj [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+    <TargetFramework>net7.0</TargetFramework>
+    <IsPackable>false</IsPackable>
+    <SonarUnknownProjectType>true</SonarUnknownProjectType>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <PackageReference Include="Microsoft.NET.Test.Sdk" Version="17.3.2" />
+    <PackageReference Include="MSTest.TestAdapter" Version="2.2.10" />
+    <PackageReference Include="MSTest.TestFramework" Version="2.2.10" />
+    <PackageReference Include="coverlet.collector" Version="3.1.2" />
```
Comment (csaba-sagi-sonarsource) [inline analyzers/its/sources/SonarLintExclusions/SonarLintExclusionsTest/SonarLintExclusionsTest.csproj:13]: Let's do a minimal example. I think the coverlet collector reference is not needed.

File: analyzers/its/sources/SonarLintExclusions/global.json [hunk 3]
```diff
@@ -0,0 +1,6 @@
+{
+  "sdk": {
+    "version": "7.0.100-rc.1.22431.12",
```
Comment (csaba-sagi-sonarsource) [inline analyzers/its/sources/SonarLintExclusions/global.json:3]: We can target the released version of .NET7 no need to use the release candidate.

File: analyzers/its/sources/SonarLintExclusions/SonarLintExclusions/NotIncluded.cs [hunk 4]
```diff
@@ -0,0 +1,12 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
```
Comment (cristian-ambrosini-sonarsource) [inline analyzers/its/sources/SonarLintExclusions/SonarLintExclusions/NotIncluded.cs:5]: Unused usings

File: analyzers/its/config/SonarLintExclusions/SonarLint.xml [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+<AnalysisInput>
+    <Rules>
+    </Rules>
+    <Settings>
+        <Setting>
+            <Key>sonar.cs.ignoreHeaderComments</Key>
+            <Value>true</Value>
+        </Setting>
+        <Setting>
+            <Key>sonar.vbnet.ignoreHeaderComments</Key>
```
Comment (cristian-ambrosini-sonarsource) [inline analyzers/its/config/SonarLintExclusions/SonarLint.xml:11]: I think that both of the properties should refer to a single language
```suggestion
            <Key>sonar.cs.ignoreHeaderComments</Key>
```

File: analyzers/its/config/SonarLintExclusions/SonarLint.xml [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+            <Value>true</Value>
+        </Setting>
+        <Setting>
+            <Key>sonar.inclusions</Key>
+            <Value>**/Included/*.cs</Value>
+        </Setting>
+        <Setting>
+            <Key>sonar.exclusions</Key>
+            <Value>**/ExcludedByExclusion.cs</Value>
```
Comment (cristian-ambrosini-sonarsource) [inline analyzers/its/config/SonarLintExclusions/SonarLint.xml:20]: Open question: is it worth adding a test for multiple exclusions/inclusions? For example, adding a new `ExcludedBySecondExclusion.cs` file and having the comma-separated list here
```suggestion
            <Value>**/ExcludedByExclusion.cs,**/ExcludedBySecondExclusion.cs</Value>
```

**Review by cristian-ambrosini-sonarsource — CHANGES_REQUESTED** [review]
Looks good! Minor improvements to discuss

File: analyzers/its/config/SonarLintExclusions/SonarLint.xml (hunk 5 above)
Comment (mary-georgiou-sonarsource) [reply analyzers/its/config/SonarLintExclusions/SonarLint.xml:11]: I see that this is how it is in all the SonarLint.xml config - I'll drop a question to understand if the vbnet case is really needed.

**Review by cristian-ambrosini-sonarsource — APPROVED** [review]
LGTM!

File: analyzers/its/config/SonarLintExclusions/SonarLint.xml (hunk 5 above)
Comment (mary-georgiou-sonarsource) [reply analyzers/its/config/SonarLintExclusions/SonarLint.xml:11]: We can leave as is - that's how it is in the SonarLint.xml

**Pull request merged**